apps/src/code-studio/progressRedux.js

```diff
@@ -397,11 +415,21 @@ export const clearProgress = () => ({
   type: CLEAR_PROGRESS
 });
 
+export const useDbProgress = () => {
```

Comment:
my understanding based on all the reading i've been doing in response to your feedback on my redux PR is that it's not good practice to perform side-effects in synchronous action creators. rather, you should leverage `redux-thunk` to have the action creator return a thunk instead of a pure action, so your side-effect will be triggered as part of the action dispatch.

from [the redux docs](https://redux.js.org/basics/actions#action-creators): "Action creators are exactly that—functions that create actions."

from [the redux-thunk docs](https://github.com/reduxjs/redux-thunk#why-do-i-need-this): "Thunks are the recommended middleware for basic Redux side effects logic"

and in [this article](https://goshakkk.name/redux-side-effect-approaches/) they have a whole section on handling side effects "inside action creators", but even there they are using the thunk approach.

i have not seen a single example of a synchronous action creator that causes side-effects. and i think that makes sense. imagine some code were gathering an array of actions to dispatch at some later time. it would call your action creator to get the `USE_DB_PROGRESS` action, and unknowingly trigger a side-effect in the process.

in this case the change is trivial:
```
return dispatch => {
  clientState.clearProgress();
  dispatch({type: USE_DB_PROGRESS})
}
```

Reply:
Oh cool!! Thank you! I'll look through those docs and update here.

Reply:
👍